api/proto/config/devmodel.proto

```diff
@@ -171,3 +156,152 @@ message PhysicalIO {
   //    For example in WWAN to which firmware version to load etc
   map <string, string> cbattr = 8;
 }
+
+// VlanAdapter represents a single VLAN sub-interface.
+// The parent, which is referenced by lower_layer_name,
+// should be either PhysicalIO or BondAdapter.
+message VlanAdapter {
+  // Logical name of this VLAN adapter.
+  string logicallabel = 1;
+
+  // A physical name of the VLAN sub-interface.
+  // Note that the interface name is limited in Linux kernel to 15 characters.
+  // If not defined, logicallabel will be used instead.
+  string interface_name = 2;
+
+  // Logical name of the lower layer adapter (bond or physicalIO).
+  string lower_layer_name = 3;
+
+  // VLAN ID. Valid values are from 1 to 4094.
+  uint32 vlan_id = 4;
+}
+
+// A bonding mode specifies the policy indicating how bonding slaves are used
+// during network transmission.
+enum BondMode {
```

Comment:
Don't some of these modes make assumptions about the MAC address assigned to the bond vs. the MAC addresses of the underlying adapters?
If they do are we sure it makes sense being able to support all of them?

Also, what is the intended default if the API does not specify a value? It would be good to document that.

Reply:
The MAC address of the bonding device is taken from its first slave device. Alternatively, it can be manually configured (via netlink or ip command). In both cases this MAC address is then passed to all the slaves and remains persistent (even if the first slave is removed) until the bonding device is brought down or reconfigured. When a bond is removed, the original MAC addresses of the slaves are restored. From the documentation and experimentation it seems that this behavior is the same in all the modes. I don't see any conflicts that this could create with the EVE's networking so we should be good to support all the modes.

Also, I have added comment to BOND_MODE_UNSPECIFIED mentioning that Round-Robin is the default mode (as it is in Linux).